test/csuite/wt2719_reconfig/main.c

```diff
@@ -0,0 +1,279 @@
+	",shared_cache=(quota=5MB)",
+	",shared_cache=(quota=0)",
+	",shared_cache=(reserve=20MB)",
+	",shared_cache=(reserve=30MB)",
+	",shared_cache=(reserve=5MB)",
+	",shared_cache=(reserve=0)",
+	",shared_cache=(size=100MB)",
+	",shared_cache=(size=1GB)",
+	",shared_cache=(size=75MB)",
+
+	",statistics=(\"all\")",
+	",statistics=(\"fast\")",
+	",statistics=(\"none\")",
+	",statistics=(\"clear\")",
+
+	",statistics_log=(json=0)",
+	",statistics_log=(json=1)",
+	",statistics_log=(on_close=0)",
+	",statistics=(\"fast\"),statistics_log=(on_close=1)",
+	",statistics_log=(sources=(\"file:\"))",
+	",statistics_log=(sources=())",
+	",statistics_log=(timestamp=\"%b:%S\")",
+	",statistics_log=(timestamp=\"%H:%M\")",
+	",statistics_log=(wait=60)",
+	",statistics_log=(wait=76)",
+	",statistics_log=(wait=37)",
+	",statistics_log=(wait=0)",
+
+	",verbose=(\"api\")",
+	",verbose=(\"block\")",
+	",verbose=(\"checkpoint\")",
+	",verbose=(\"compact\")",
+	",verbose=(\"evict\")",
+	",verbose=(\"evictserver\")",
+	",verbose=(\"fileops\")",
+	",verbose=(\"handleops\")",
+	",verbose=(\"log\")",
+	",verbose=(\"lsm\")",
+	",verbose=(\"lsm_manager\")",
+	",verbose=(\"metadata\")",
+	",verbose=(\"mutex\")",
+	",verbose=(\"overflow\")",
+	",verbose=(\"read\")",
+	",verbose=(\"rebalance\")",
+	",verbose=(\"reconcile\")",
+	",verbose=(\"recovery\")",
+	",verbose=(\"salvage\")",
+	",verbose=(\"shared_cache\")",
+	",verbose=(\"split\")",
+	",verbose=(\"temporary\")",
+	",verbose=(\"transaction\")",
+	",verbose=(\"verify\")",
+	",verbose=(\"version\")",
+	",verbose=(\"write\")",
+	",verbose=()"
+};
+
+static int
+handle_message(WT_EVENT_HANDLER *handler,
+    WT_SESSION *session, const char *message)
+{
+	(void)(handler);
+	(void)(session);
+	(void)(message);
+
+	/* We configure verbose output, so just ignore. */
+	return (0);
+}
+
+static WT_EVENT_HANDLER event_handler = { NULL, handle_message, NULL, NULL };
+
+int
+main(int argc, char *argv[])
+{
+	enum { CACHE_SHARED, CACHE_SET, CACHE_NONE } cache;
+	TEST_OPTS *opts, _opts;
+	WT_RAND_STATE rnd;
+	WT_SESSION *session;
+	size_t len;
+	u_int i, j;
+	int ret;
+	const char *p;
+	char *config;
+
+	opts = &_opts;
+	memset(opts, 0, sizeof(*opts));
+	opts->table_type = TABLE_ROW;
+	testutil_check(testutil_parse_opts(argc, argv, opts));
+	testutil_make_work_dir(opts->home);
+
+	testutil_check(
+	    wiredtiger_open(opts->home, &event_handler, "create", &opts->conn));
+
+	/* Open an LSM file so the LSM reconfiguration options make sense. */
+	testutil_check(
+	    opts->conn->open_session(opts->conn, NULL, NULL, &session));
+	testutil_check(session->create(
+	    session, opts->uri, "type=lsm,key_format=S,value_format=S"));
+
+	/*
+	 * Initialize the global RNG. Start with the standard seeds, and then
+	 * use seconds since the Epoch modulo a prime to run the RNG for some
+	 * number of steps, so we don't start with the same values every time.
+	 */
+	__wt_random_init(&rnd);
+	for (i = (u_int)time(NULL) % 10007; i > 0; --i)
```

Comment:
@keithbostic this use of our random number generator bothers me - can we use {{\__wt_random_init_seed}} instead? The way the seeding works for our random number generator is to effectively step through the sequence in a more random way than this. i.e: it randomly steps the independent {{W}} and {{Z}} variables forward through their sequence. It does not alter the randomness of the generator.